tensorflow/core/lib/strings/str_util.cc

```diff
@@ -416,6 +417,100 @@ bool ConsumeNonWhitespace(StringPiece* s, StringPiece* val) {
   }
 }
 
+Status SplitUTF8(StringPiece text, const string& delim, const bool skip_empty,
+                 std::vector<string>* result) {
+  // Bytes    Byte 1    Byte 2    Byte 3    Byte 4
+  //   1     0xxxxxxx
+  //   2     110xxxxx  10xxxxxx
+  //   3     1110xxxx  10xxxxxx  10xxxxxx
+  //   4     11110xxx  10xxxxxx  10xxxxxx  10xxxxxx
+  result->clear();
+  size_t char_start = 0, char_len = 0;
+  bool last_char_was_a_delimiter = true;
+  for (size_t i = 0; i < text.size(); ++i) {
+    if (char_start == i) {
+      if (!UTF8CharNumBytes(text, &char_len)) {
+        result->clear();
+        return errors::InvalidArgument("Invalid UTF8 encoding at byte ", i);
+      }
+    }
+    // Validate all intermediate bytes are valid UTF-8
+    if (TF_PREDICT_FALSE((char_start != i) && (text[i] & 0xC0) != 0x80)) {
+      return errors::InvalidArgument("Invalid UTF8 encoding at byte ", i);
+    }
+    // Reached the end of a character
+    if (char_start + char_len == i + 1) {
+      const StringPiece entry = text.substr(char_start, char_len).ToString();
+      if (delim == "") {
+        result->emplace_back(entry.ToString());
+      } else if (delim == entry) {
+        if (last_char_was_a_delimiter && !skip_empty) {
+          result->emplace_back("");
+        }
+        last_char_was_a_delimiter = true;
+        if (i + 1 == text.size() && !skip_empty) {
+          // If this is the last we always add an empty one
+          // to handle split(",",",") -> ["", ""]
+          result->emplace_back("");
+        }
+      } else if (last_char_was_a_delimiter) {
+        last_char_was_a_delimiter = false;
+        result->emplace_back(entry.ToString());
+      } else if (result->size() == 0) {
```

Comment:
@asimshankar This condition might actually be needed. The reason is that the following line 
```
      } else {
        result->back() = result->back() + entry.ToString();
      }
```
might have result empty. So incase result is empty then we will initialize instead of append to `result->back()`.